crates/miden-lib/asm/miden/input_note.masm

```diff
@@ -141,3 +151,121 @@ export.get_metadata
     swapdw dropw dropw swapw dropw
     # => [METADATA]
 end
+
+#! Returns the the inputs commitment and length of the input note with the specified index.
+#!
+#! Inputs:  [note_index]
+#! Outputs: [NOTE_INPUTS_COMMITMENT, num_inputs]
+#!
+#! Where:
+#! - note_index is the index of the input note whose data should be returned.
+#! - NOTE_INPUTS_COMMITMENT is the inputs commitment of the specified input note.
+#! - num_inputs is the number of input values of the specified input note.
+#!
+#! Panics if:
+#! - the note index is greater or equal to the total number of input notes.
+#!
+#! Invocation: exec
+export.get_data_info
```

Comment:
```suggestion
export.get_payload_info
```

We'd also have to rename `num_inputs` to `payload_size` or `payload_len` and `NOTE_INPUTS_COMMITMENT` into `NOTE_PAYLOAD_COMMITMENT`.